api/src/main/java/bio/terra/tanagra/service/search/Selection.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.tanagra.service.search;
+
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+
+/**
+ * A construct in a query that represents a value to be selected.
+ *
+ * <p>Represents a single column in a SELECT SQL clause.
+ */
+public interface Selection {
```

Comment:
Is there a clear path for "group by" in this model?

Reply:
Not yet. In many cases, I think we only care about any grouping by at the top most query level, but it's feasible to consider using group by's within a filter.